sys/Makefile.include

```diff
@@ -100,7 +100,9 @@ endif
 
 ifneq (,$(filter printf_float,$(USEMODULE)))
   ifneq (,$(filter newlib_nano,$(USEMODULE)))
-    LINKFLAGS += -u _printf_float
+    ifeq (,$(filter mpaland-printf,$(USEPKG)))
+      LINKFLAGS += -u _printf_float
+    endif
```

Comment:
Ooof, at now three levels deep of un-intuitive if statements, I think this bit of code deserves a comment to describe its intent.

Reply:
I added some comments explaining why and when the linker flags are needed.